janusgraph-es/src/test/java/org/janusgraph/diskstorage/es/ElasticSearchConfigTest.java

```diff
@@ -227,15 +233,48 @@ public void testExternalDynamic() throws Exception {
             idx.register(storeName, "date", IndexProviderTest.getMapping(idx.getFeatures(), ANALYZER_ENGLISH, ANALYZER_KEYWORD).get("date"), itx);
             fail("should fail");
         } catch (final PermanentBackendException e) {
+            log.debug(e.getMessage(), e);
+        }
+
+        final HttpPut newMapping = new HttpPut("janusgraph_"+storeName);
+        newMapping.setEntity(new StringEntity(objectMapper.writeValueAsString(readMapping(version, "/dynamic_mapping.json")), Charset.forName("UTF-8")));
+        executeRequest(newMapping);
+
+        // Test that the "date" property works well.
+        idx.register(storeName, "date", IndexProviderTest.getMapping(idx.getFeatures(), ANALYZER_ENGLISH, ANALYZER_KEYWORD).get("date"), itx);
+        // Test that the "weight" property works well due to external mapping
+        idx.register(storeName, "weight", IndexProviderTest.getMapping(idx.getFeatures(), ANALYZER_ENGLISH, ANALYZER_KEYWORD).get("weight"), itx);
+        itx.rollback();
+        idx.close();
+    }
+
+    @Test
+    public void testExternalDynamicWithDynamicTemplate() throws Exception {
+        final Duration maxWrite = Duration.ofMillis(2000L);
+        final String storeName = "test_mapping";
+        final Configuration indexConfig = GraphDatabaseConfiguration.buildGraphConfiguration().set(USE_EXTERNAL_MAPPINGS, true, INDEX_NAME).set(USE_DYNAMIC_TEMPLATE, true, INDEX_NAME).restrictTo(INDEX_NAME);
+        final IndexProvider idx = open(indexConfig);
+        final ElasticMajorVersion version = ((ElasticSearchIndex) idx).getVersion();
+
+        //Test create index KO mapping is not push
```

Comment:
Please clarify what this means: what's a "KO mapping"?

Also, additional comments:

* "is not push" -> "is not pushed"
* space after `//`
* period at the end of the sentence, please